objectivec/GPBCodedInputStream.m

```diff
@@ -149,6 +120,10 @@ static int64_t ReadRawVarint64(GPBCodedInputStreamState *state) {
   return 0;
 }
 
+static int32_t ReadRawVarint32(GPBCodedInputStreamState *state) {
+  return (int32_t)ReadRawVarint64(state);
```

Comment:
Can you double check that there is a test that ensure truncation is happening as expected (both positive and negative values)